code/modules/cargo/supplypod.dm

```diff
@@ -0,0 +1,313 @@
+	var/list/explosionSize = list(0,0,2,3)
+
+/obj/structure/closet/supplypod/bluespacepod
+	icon_state = "bluespacepod"
+	bluespace = TRUE
+	explosionSize = list(0,0,1,2)
+	landingDelay = 30
+
+/obj/structure/closet/supplypod/centcompod
+	icon_state = "centcompod"
+	bluespace = TRUE
+	explosionSize = list(0,0,0,0)
+	landingDelay = 5
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+
+/obj/structure/closet/supplypod/Initialize()
+	. = ..()
+	icon_state_to_name_desc()//set the name and description based on the icon state.
+
+/obj/structure/closet/supplypod/update_icon()
+	cut_overlays()
+	if (icon_state)//check to ensure pod isnt "invisible"
+		if (opened)
+			add_overlay("[icon_state]_open")
+		else
+			add_overlay("[icon_state]_door")
+
+/obj/structure/closet/supplypod/tool_interact(obj/item/W, mob/user)
+	if (bluespace)
+		return FALSE
+	else
+		..()
+
+/obj/structure/closet/supplypod/ex_act()
+	return
+
+/obj/structure/closet/supplypod/contents_explosion()
+	return
+
+/obj/structure/closet/supplypod/toggle(mob/living/user)
+	return
+
+/obj/structure/closet/supplypod/proc/preOpen()
+	var/turf/T = get_turf(src)
+	var/boomTotal = 0
+	for (var/mob/living/M in T)
+		if (effectLimb && iscarbon(M))
+			var/mob/living/carbon/CM = M
+			for (var/obj/item/bodypart/bodypart in CM.bodyparts)
+				if(bodypart.body_part != HEAD && bodypart.body_zone != CHEST)//we dont want to kill him, just teach em a lesson!
+					bodypart.dismember()
+					playsound(CM,pick('sound/misc/desceration-01.ogg','sound/misc/desceration-02.ogg','sound/misc/desceration-01.ogg') ,50, 1, -1)
+		if (effectGib)
+			M.gib()
+			damage=5000
+		M.adjustBruteLoss(damage)
+	var/list/B = explosionSize
+	for (var/i in 1 to B.len-1)
+		boomTotal += i
+	if (boomTotal != 0)
+		explosion(get_turf(src), B[1], B[2], B[3], flame_range = B[4], silent = effectQuiet, ignorecap = istype(src, /obj/structure/closet/supplypod/centcompod)) //less advanced equipment than bluespace pod, so larger explosion when landing
+	else if (!effectQuiet)
+		playsound(src, "explosion", 80, 1)//theres no explosion, but we can still pretend!
+	if (damage != 0)
+		for(var/mob/living/M in T.contents)
+			M.adjustBruteLoss(damage)
+	if (effectMissile)
+		opened = TRUE
+		qdel(src)
+	if (icon_state == "gondolapod")
+		var/mob/living/simple_animal/pet/gondola/gondolapod/benis = new(get_turf(src), src)
+		benis.contents |= contents
+		moveToNullspace()
+		addtimer(CALLBACK(src, .proc/open, benis), openingDelay)//open 3seconds after appearing
+	else
+		addtimer(CALLBACK(src, .proc/open, src), openingDelay)//open 3seconds after appearing
+
+/obj/structure/closet/supplypod/open(atom/movable/holder, var/broken = FALSE, var/manual = FALSE)
+	var/turf/T = get_turf(holder)
+	var/mob/M
+	if (istype(holder, /mob))
+		M = holder
+		if (M.key && !manual && !broken)//dont auto-open if we are player controlled (usually a gondola), 
+			return
+	opened = TRUE
+	INVOKE_ASYNC(holder, .proc/setOpened)

```

Comment:
why async

Reply:
frequently this foolish fruit fails to fathom my fantastic functions

Reply:
this means i have no fucking idea why i did that

Reply:
then do not

Reply:
@optimumtact wait i remember why i did
it was so i could call the setOpened proc on the atom/movable/holder (in our case the gondolapod mob) without having to code a setOpened proc into the atom class

Reply:
i have a suspiciion that doing this was retarded what do u think